EventPlanner/app/src/main/java/m3/eventplanner/fragments/event/EventDetailsFragment.java

```diff
@@ -0,0 +1,139 @@
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.squareup.picasso.Picasso;
+
+import java.util.Collection;
+import java.util.List;
+
+import m3.eventplanner.R;
+import m3.eventplanner.adapters.AgendaItemListAdapter;
+import m3.eventplanner.auth.TokenManager;
+import m3.eventplanner.clients.ClientUtils;
+import m3.eventplanner.databinding.FragmentEventDetailsBinding;
+import m3.eventplanner.models.AddFavouriteEventDTO;
+import m3.eventplanner.models.GetAgendaItemDTO;
+import m3.eventplanner.models.GetEventDTO;
+import m3.eventplanner.models.GetOrganizerDTO;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class EventDetailsFragment extends Fragment {
+    private FragmentEventDetailsBinding binding;
+    private EventDetailsViewModel viewModel;
+    private ClientUtils clientUtils;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = FragmentEventDetailsBinding.inflate(inflater, container, false);
+        return binding.getRoot();
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        viewModel = new ViewModelProvider(this).get(EventDetailsViewModel.class);
+        clientUtils = new ClientUtils(requireContext());
+        viewModel.initialize(clientUtils);
+
+        setupObservers();
+        setupClickListeners();
+
+        // Load initial data
+        if (getArguments() != null) {
+            int eventId = getArguments().getInt("selectedEventId");
+            int accountId = new TokenManager(requireContext()).getAccountId();
+            if(accountId==0)
+                binding.favouriteButton.setVisibility(View.GONE);
+            viewModel.loadEventDetails(eventId, accountId);
+        }
+    }
+
+    private void setupObservers() {
+        viewModel.getEvent().observe(getViewLifecycleOwner(), this::populateEventDetails);
+
+        viewModel.getAgenda().observe(getViewLifecycleOwner(), agendaItems -> {
+            binding.agendaRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+            binding.agendaRecyclerView.setAdapter(new AgendaItemListAdapter(agendaItems));
+        });
+
+        viewModel.getIsFavourite().observe(getViewLifecycleOwner(), isFavourite ->
+                binding.favouriteButton.setImageResource(isFavourite ?
+                        R.drawable.heart_filled : R.drawable.heart_outline)
+        );
+
+        viewModel.getError().observe(getViewLifecycleOwner(), error ->
+                Toast.makeText(getContext(), error, Toast.LENGTH_SHORT).show()
```

Comment:
Maybe we should consider using Snackbar instead of Toast, like in frontend. It is more visible and can include an action.